src/lib.rs

```diff
@@ -1690,4 +1713,39 @@ mod tests {
         assert_ne!(map_a, map_c);
         assert_ne!(map_c, map_a);
     }
+
+
+    #[test]
+    fn drain_basic() {
+        let mut map_a = OrderMap::new();
```

Comment:
I've replaced the example unit tests with a drain quickcheck test. This is my first time using Rust's quickcheck, please let me know if you think I made some newbie error.